pipelines/batch/src/main/java/org/openmrs/analytics/ResourceTag.java

```diff
@@ -32,4 +34,34 @@ public class ResourceTag implements Serializable {
   Coding coding;
   String resourceId;
   Integer tagType;
+
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    }
+
+    if (!(other instanceof ResourceTag)) {
+      return false;
+    }
+
+    ResourceTag resourceTag = (ResourceTag) other;
+
+    return coding != null
+        && coding.equalsDeep(resourceTag.coding)
+        && StringUtils.compare(resourceId, resourceTag.getResourceId()) == 0
+        && ObjectUtils.compare(tagType, resourceTag.getTagType()) == 0;
+  }
+
+  @Override
+  public int hashCode() {
+    int h$ = 1;
```

Comment:
nit: Why not just call this `hash` or even `h`; what does `$` convey? From [the spec](https://docs.oracle.com/javase/specs/jls/se11/html/jls-3.html#jls-3.8): _"The dollar sign should be used only in mechanically generated source code or, rarely ..."_.

Reply:
done